### PR TITLE
add a new case virt_firmware_basic_test

### PR DESCRIPTION
**Pull request by XueqiangWei**
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 2129169

### CODE REVIEW

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (13.67 s)

**Comment by XueqiangWei** [comment]
@yanan-fu Please help review it. Many thanks.

File: qemu/tests/virt_firmware_basic_test.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                                             shell=True)
+    if status:
+        test.error("Failed to clone the virt-firmware repo,"
+                   "the error message is '%s'." % output)
+    test_dirname = os.path.join(virt_firmware_dirname, "tests")
+    for file_name in os.listdir(test_dirname):
+        if re.search(test_file_pattern, file_name, re.I):
+            test_file = os.path.join(test_dirname, file_name)
+            if file_name.endswith("py"):
+                test_cmd = params["python_cmd"] % test_file
```
Comment (yanan-fu) [inline qemu/tests/virt_firmware_basic_test.py:59]: Suggest using `virttest.utils_test.find_python` to get the python bin.

Comment (XueqiangWei) [reply qemu/tests/virt_firmware_basic_test.py:59]: Thanks Yanan. This funciton is used for look for the python binary installed in the guest. For this case, run the tests in host.

File: qemu/tests/virt_firmware_basic_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    check_virt_firmware_cmd = params["check_virt_firmware_cmd"]
+    status = process.system(check_virt_firmware_cmd,
+                            ignore_status=True,
+                            shell=True)
+    if status:
+        test.cancel("Please install python-virt-firmware on the host.")
```
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/virt_firmware_basic_test.py:40]: Thank you Yanan, this way is simpler, I will update it.

File: qemu/tests/cfg/virt_firmware_basic_test.cfg [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- virt_firmware_basic_test:
+    only q35
+    only ovmf
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8 Host_RHEL.m9.u0 Host_RHEL.m9.u1
+    virt_test_type = qemu
+    type = virt_firmware_basic_test
+    start_vm = no
+    check_virt_firmware_cmd = "which virt-fw-vars"
```
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/cfg/virt_firmware_basic_test.cfg:8]: updated, thanks.

File: qemu/tests/virt_firmware_basic_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    virt_firmware_dirname = params["virt_firmware_dirname"]
+    virt_firmware_repo_addr = params["virt_firmware_test_repo"]
+    clone_cmd = params["clone_cmd"]
+    test_file_pattern = params["test_file_pattern"]
+    if os.path.exists(virt_firmware_dirname):
+        shutil.rmtree(virt_firmware_dirname, ignore_errors=True)
+    clone_cmd %= (virt_firmware_repo_addr, virt_firmware_dirname)
+    status, output = process.getstatusoutput(clone_cmd,
```
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/virt_firmware_basic_test.py:48]: updated, thanks.

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (11.73 s)

**Comment by XueqiangWei** [comment]
@yanan-fu Updated, please help review it again. Many thanks.

File: qemu/tests/virt_firmware_basic_test.py (hunk 1 above)
Comment (yanan-fu) [reply qemu/tests/virt_firmware_basic_test.py:53]: Yeah, you are right.   Then we can use `command -v python python3 | head -1` for it.

**Comment by yanan-fu** [comment]
@XueqiangWei  How about the test script's output ?  I don't know what does it looks like now,  do we need pay attention to it,  such as be used to debug the failed scenario ?

File: qemu/tests/virt_firmware_basic_test.py (hunk 1 above)
Comment (XueqiangWei) [reply qemu/tests/virt_firmware_basic_test.py:53]: Thanks Yanan, updated.

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (19.15 s)

**Comment by XueqiangWei** [comment]
@yanan-fu Updated, please help review it. Many thanks.

File: qemu/tests/virt_firmware_basic_test.py (hunk 1 above)
Comment (PaulYuuu) [reply qemu/tests/virt_firmware_basic_test.py]: `sys.executable`

Comment (XueqiangWei) [reply qemu/tests/virt_firmware_basic_test.py]: Thanks Yihuang, I will update it.

**Comment by XueqiangWei** [comment]
Test result:
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (17.53 s)

**Comment by XueqiangWei** [comment]
@yanan-fu @PaulYuuu Updated, please help review it again. Many thanks.

File: qemu/tests/cfg/virt_firmware_basic_test.cfg [hunk 5]
```diff
@@ -0,0 +1,10 @@
+- virt_firmware_basic_test:
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8 Host_RHEL.m9.u0 Host_RHEL.m9.u1
```
Comment (yanan-fu) [inline qemu/tests/cfg/virt_firmware_basic_test.cfg:2]: `no i440fx` is make sense I think.

Comment (XueqiangWei) [reply qemu/tests/cfg/virt_firmware_basic_test.cfg:3]: Agree with you. Updated it, please help take a look at it. Many thanks.
Test result:
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (18.60 s)

**Review by hereischen — APPROVED** [review]

File: qemu/tests/cfg/virt_firmware_basic_test.cfg [hunk 6]
```diff
@@ -0,0 +1,11 @@
+- virt_firmware_basic_test:
+    no i440fx
```
Comment (PaulYuuu) [inline qemu/tests/cfg/virt_firmware_basic_test.cfg:2]: Actually, only x86 and aarch64 can use edk2, so this is another limitation.

File: qemu/tests/cfg/virt_firmware_basic_test.cfg [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    no Host_RHEL.m6 Host_RHEL.m7 Host_RHEL.m8 Host_RHEL.m9.u0 Host_RHEL.m9.u1
```
Comment (PaulYuuu) [inline qemu/tests/cfg/virt_firmware_basic_test.cfg:3]: However, virt-firmware can be installed by users using [pip install virt-firmware](https://pypi.org/project/virt-firmware/), so I think this line is not suitable, the virt-firmware should not link to any distro.

File: qemu/tests/cfg/virt_firmware_basic_test.cfg (hunk 6 above)
Comment (XueqiangWei) [reply qemu/tests/cfg/virt_firmware_basic_test.cfg:2]: Thanks Yihuang, I will update it.

**Comment by XueqiangWei** [comment]
(1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.virt_firmware_basic_test.q35: PASS (11.72 s)

**Comment by XueqiangWei** [comment]
@yanan-fu @PaulYuuu Updated, please help review it again. Many thanks.

**Review by yanan-fu — APPROVED** [review]
Ack

**Review by PaulYuuu — APPROVED** [review]
LGTM

**Pull request merged**